onnxruntime/python/tools/quantization/quant_utils.py

```diff
@@ -109,29 +109,46 @@ def quantize_nparray(qType, arr, scale, zero_point, low=None, high=None):
     return arr_fp32.astype(dtype)
 
 
-def compute_scale_zp(rmin, rmax, qType, quantize_range, symmetric):
-    if qType == onnx_proto.TensorProto.INT8:
-        if symmetric:
-            max_range = max(abs(rmin), abs(rmax))
-            scale = (float(max_range) * 2) / quantize_range if max_range > 0 else 1.0
-            zero_point = 0
-        else:
-            max_range = float(rmax) - float(rmin)
-            scale = float(max_range) / quantize_range if max_range > 0 else 1.0
-            zero_point = round((quantize_range / 2) - rmax / scale)
-    elif qType == onnx_proto.TensorProto.UINT8:
-        scale = (float(rmax) - rmin) / quantize_range if rmin != rmax else 1
-        zero_point = round((0 - rmin) / scale)  # round to nearest integer
-    else:
-        raise ValueError("Unexpected data type {} requested. Only INT8 and UINT8 are supported.".format(qType))
+def compute_scale_zp(rmin, rmax, qmin, qmax, symmetrize=False):
+    '''
+    Calculate the scale s and zero point z for the quantization relation 
+    r = s(q-z), where r are the original values and q are the corresponding
+    quantized values. 
+
+    r and z are calculated such that every value within [rmin,rmax] has an
+    approximate representation within [qmin,qmax]. In addition, qmin <= z <=
+    qmax is enforced. If the symmetrize flag is set to True, the interval
+    [rmin,rmax] is symmetrized to [-absmax, +absmax], where
+    absmax = max(abs(rmin), abs(rmax)).
+
+    :parameter rmin: minimum value of r
+    :parameter rmax: maximum value of r
+    :parameter qmin: minimum value representable by the target quantization data type
+    :parameter qmax: maximum value representable by the target quantization data type
+    :return: zero and scale [z, s]
+
+    '''
+    
+    # Adjust rmin and rmax such that 0 is included in the range. This is
+    # required to make sure zero can be represented by the quantization data
+    # type (i.e. to make sure qmin <= zero_point <= qmax)
+    rmin = min(rmin, 0)
+    rmax = max(rmax, 0)
+
+    if symmetrize:
+        absmax = max(abs(rmin), abs(rmax))
+        rmin = -absmax
+        rmax = +absmax
+
+    scale = (rmax - rmin) / float(qmax-qmin) if rmax!=rmin else 1.0
+    zero_point = round(qmin - rmin/scale)
 
     return [zero_point, scale]
 
 
-def quantize_data(data, quantize_range, qType, symmetric=True):
+def quantize_data(data, qType, symmetrize, reduce_range=False):
```

Comment:
>symmetrize [](http://example.com/codeflow?start=31&length=10)

could you please use symmetric to keep consistent?

Reply:
Done